File: Program's_Contributed_By_Contributors/Python_Programs/sort012.py
def sort012(array):
  low=0
  high=len(array)-1
  mid=0
  while mid<=high:
    if array[mid]==0:
      array[low],array[mid]=array[mid],array[low]
      low=low+1
      mid=mid+1
    elif array[mid]==1:
      mid=mid+1
    else:
      array[mid],array[high]=array[high],array[mid]
      high=high-1
  return array

arr=[0, 1, 1, 0, 1, 2, 1, 2, 0, 0, 0, 1]
ans = sort012(arr)
print(ans)

File: DSA problems in c++/sort012.cpp
#include<iostream>
using namespace std;

void sortArray(int arr[],int n)
{
    int low = 0;
    int mid = 0;
    int high = n - 1;

    while(mid <= high)
    {
        if(arr[mid] == 0)
        {
            swap(arr[low],arr[mid]);
            low++;
            mid++;
        }

        else if(arr[mid] == 1)
        {
            mid++;
        }
        else if(arr[mid] == 2)
        {
            swap(arr[mid],arr[high]);
            high--;
        }
        
    }
}

void printArray(int arr[],int n)
{
    for(int i = 0; i < n; i++)
    {
        cout << arr[i] <<" ";
    }
    cout << endl;
}

int main()
{
    int arr[6] = {0,1,2,2,1,0};

    sortArray(arr,6);
    printArray(arr,6); 


    return 0;
}